pyerrors/correlators.py

```diff
@@ -1081,7 +1066,7 @@ def _sort_vectors(vec_set, ts):
 
 
 def _GEVP_solver(Gt, G0):  # Just so normalization an sorting does not need to be repeated. Here we could later put in some checks
-    sp_val, sp_vecs = scipy.linalg.eig(Gt, G0)
+    sp_val, sp_vecs = scipy.linalg.eigh(Gt, G0)
     sp_vecs = [sp_vecs[:, np.argsort(sp_val)[-i]] for i in range(1, sp_vecs.shape[0] + 1)]
```

Comment:
I think the ordering is implicitly done in the `eigh` call and is redundant. Can you confirm that?

Reply:
This is a little more involved. First of all, `eigh` sorts in descending order, but in lattice applications, you want to have the lowest states. So you have to invert this. On top of this, a matrix of eigenvectors is returned and you need to get the columns of this matrix, in order to have the eigenvectors. 
So you go from 
```
array([[ 4.08809560e+01,  4.71498284e+00,  6.77498937e-01,
        -1.66951604e+01],
       [-4.58757541e+02, -5.50717440e+02, -3.16503614e+01,
         1.99051675e+03],
       [ 5.38668672e+02,  5.26371852e+02,  7.59161722e+01,
        -5.84411783e+03],
       [-2.53763739e+01,  4.70594189e+02,  2.34206426e+02,
         4.96899729e+03]])
```
to 
```
[array([  -16.69516037,  1990.51675171, -5844.11782914,  4968.99728957]),
 array([  0.67749894, -31.65036143,  75.91617222, 234.20642623]),
 array([   4.71498284, -550.71744041,  526.37185249,  470.59418928]),
 array([  40.88095603, -458.75754148,  538.66867207,  -25.37637386])]
```
(based on the data in the example. Is this correct, @JanNeuendorf?

Reply:
In general, this functions needs a doc-string (although it is an internal function). The order of the returned eigenvectors should be specified. Also, we need to specify that the results are incorrect, if the matrices are non-symmetric or explicitly catch those cases.